googlecloud-plugins/src/main/java/co/cask/hydrator/plugin/source/BigQuerySource.java

```diff
@@ -0,0 +1,221 @@
+        case DOUBLE:
+          builder.set(fieldName, Double.parseDouble(map.get(fieldName)));
+          break;
+        case BOOLEAN:
+          builder.set(fieldName, Boolean.parseBoolean(map.get(fieldName)));
+          break;
+        case NULL:
+          builder.set(fieldName, null);
+          break;
+        case STRING:
+          builder.set(fieldName, map.get(fieldName));
+          break;
+        default:
+          throw new UnexpectedFormatException("field type " + fieldType + " is not supported.");
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * {@link PluginConfig} class for {@link BigQuerySource}
+   */
+  public static class BQSourceConfig extends PluginConfig {
+    private static final String PROJECTID_DESC = "The ID of the project in Google Cloud";
+    private static final String TEMP_BUCKET_DESC = "The temporary Google Cloud Storage directory to be used for " +
+                                                   "storing the intermediate results. " +
+                                                   "e.g. 'gs://bucketname/directoryname'. The directory should not " +
+                                                   "already exist. Users should manually delete this directory " +
+                                                   "afterwards to avoid any extra Google Storage charges.";
+    private static final String IMPORT_QUERY_DESC = "The SELECT query to use to import data from the specified table." +
+                                                    " For example: 'SELECT TOP(corpus, 10) as title, COUNT(*) as " +
+                                                    "unique_words FROM publicdata:samples.shakespeare', where " +
+                                                    "'publicdata' is the project name (optional), 'samples' is the " +
+                                                    "dataset name, and 'shakespare' is the table name. If this query " +
+                                                    "is not provided, reads instead from the configured inputTable.";
+    private static final String JSON_KEYFILE_DESC = "JSON key file path for credentials, Note that user should " +
+                                                    "upload his credential file to the same path of all nodes.";
+    private static final String INPUT_TABLE_DESC = "The BigQuery table to read from, in the form " +
+                                                   "'<projectId (optional)>:<datasetId>.<tableId>'. The 'projectId' " +
+                                                   "is optional. Example: 'publicdata:samples.shakespeare'. Note: if " +
+                                                   "the import query is specified, this table should already exist " +
+                                                   "and be an empty table with the query result schema.";
+    private static final String OUTPUTSCHEMA_DESC = "Comma-separated mapping of output schema column names to " +
```

Comment:
Check that all these strings match the corresponding strings above in the documentation.